### PR TITLE
test: Port measurement_names/table_names query_tests to end-to-end tests

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to https://github.com/influxdata/influxdb_iox/issues/6617.

I recommend reviewing commit-by-commit, there were a few small test helper fixes separate from the porting that are explained in each commit.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+use super::{run_data_test, InfluxRpcTest};
+use async_trait::async_trait;
+use data_types::{MAX_NANO_TIME, MIN_NANO_TIME};
+use futures::{prelude::*, FutureExt};
+use std::sync::Arc;
+use test_helpers_end_to_end::{DataGenerator, GrpcRequestBuilder, MiniCluster, StepTestState};
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:8]: This test was from influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs, not query_tests

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                    vec!["attributes", "cpu_load_short", "status", "swap", "system"]
+                );
+            }
+            .boxed()
+        }),
+    )
+    .await
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:50]: This was `list_table_names_no_data_pred` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L53-L60

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "TwoMeasurements",
+        request: GrpcRequestBuilder::new(),
+        expected_names: vec!["cpu", "disk"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:61]: This was `list_table_names_no_data_passes` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L63-L71

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        setup_name: "TwoMeasurementsManyFields",
+        // no rows pass this predicate
+        request: GrpcRequestBuilder::new().timestamp_range(10_000_000, 20_000_000),
+        expected_names: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:73]: This was `list_table_names_no_non_null_data_passes` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L74-L84

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            // model predicate like _field='reading' which last row does not have
+            .field_predicate("reading")
+            .measurement_predicate("o2"),
+        expected_names: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:91]: This was `list_table_names_no_non_null_general_data_passes` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L87-L101

The predicate seemed to be more complicated than it needed to be, and did a `> 50` query on a tag which I don't think is valid Influx? Let me know if I'm missing something.

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+            .field_predicate("reading")
+            .measurement_predicate("o2")
+            .tag_predicate("state", "CA"),
+        expected_names: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:110]: These next three tests replace these five tests: https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L104-L126

I didn't really see additional coverage being added with some of those tests, so I consolidated them into timestamp range queries that matched all measurements, some measurements, and no measurements. Let me know if I'm missing an aspect the old tests were covering.

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "TwoMeasurements",
+        request: GrpcRequestBuilder::new().timestamp_range(250, 350),
+        expected_names: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:143]: This was `list_table_names_max_time_included` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L129-L136

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "MeasurementWithMaxTime",
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME + 1, MAX_NANO_TIME + 1),
+        expected_names: vec!["cpu"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:154]: This was `list_table_names_max_time_excluded` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L139-L147

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "MeasurementWithMaxTime",
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME + 1, MAX_NANO_TIME),
+        expected_names: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:165]: This was `list_table_names_all_time` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L150-L157

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "MeasurementWithMaxTime",
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME, MAX_NANO_TIME + 1),
+        expected_names: vec!["cpu"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:176]: This was `list_table_names_with_periods` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L160-L167

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+    Arc::new(MeasurementNamesTest {
+        setup_name: "PeriodsInNames",
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME, MAX_NANO_TIME + 1),
+        expected_names: vec!["measurement.one"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:187]: As far as I can tell, after I fixed `GrpcRequestBuilder` in another commit in this PR, measurement names now supports arbitrary predicates, so I added this test to address this comment: https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/table_names.rs#L169-L171

**Review by carols10cents — COMMENTED** [review]
Matching up tests

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs (hunk 1 above)
Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:8]: Thank you for the unificiation

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs (hunk 5 above)
Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:91]: I think `temp` is a field (not a tag) in this case. The predicate was more complicated than an equality predicate to force the planner through a different plan path (but the special case path may no longer be used anyways, so this is probably ok). I'll plan on updating this PR to add the predicate back in.

File: test_helpers_end_to_end/src/grpc.rs [hunk 12]
```diff
@@ -42,11 +42,22 @@ impl GrpcLiteral for f64 {
     }
 }
 
+impl GrpcLiteral for i64 {
```
Comment (alamb) [inline test_helpers_end_to_end/src/grpc.rs:45]: 👍

**Review by alamb — APPROVED** [review]
There is the one predicate that needs a tweak, but I'll make that change as part of a follow on PR

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs (hunk 5 above)
Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_names.rs:91]: Actually, I spent some time messing around with this and I think you are right -- `temp > 50` is not supported in the grpc interface

**Comment by alamb** [comment]
Merged up to fix conflicts

**Pull request merged**